kratos/utilities/element_size_calculator.cpp

```diff
@@ -827,6 +1406,98 @@ double ElementSizeCalculator<3,4>::AverageElementSizeDerivative(
     KRATOS_CATCH("");
 }
 
+// Tetrahedra3D10 version.
+template<>
+double ElementSizeCalculator<3,10>::AverageElementSize(const Geometry<Node<3> >& rGeometry)
+{
+
+    const double x10 = rGeometry[1].X() - rGeometry[0].X();
+    const double y10 = rGeometry[1].Y() - rGeometry[0].Y();
+    const double z10 = rGeometry[1].Z() - rGeometry[0].Z();
+
+    const double x20 = rGeometry[2].X() - rGeometry[0].X();
+    const double y20 = rGeometry[2].Y() - rGeometry[0].Y();
+    const double z20 = rGeometry[2].Z() - rGeometry[0].Z();
+
+    const double x30 = rGeometry[3].X() - rGeometry[0].X();
+    const double y30 = rGeometry[3].Y() - rGeometry[0].Y();
+    const double z30 = rGeometry[3].Z() - rGeometry[0].Z();
```

Comment:
[MathUtils](https://github.com/KratosMultiphysics/Kratos/blob/a602d4c431df9a1ead0a881af9bfc4fc95cdbf0a/kratos/utilities/math_utils.h)? We have cross products, dot products, norms, determinants, etc in there too.

Reply:
These functions were implemented in this closed form manner to avoid allocating any auxiliary matrix/array. Note that these are likely called for each element at each non-linear iteration.